### PR TITLE
Bring auth to the SDK and refactor mistakes from the past

### PR DESCRIPTION
**Pull request by alex-seville**
Added methods to handle authenticating a user on node.
Removed browser support for authenticating since it's not advisable to encourage it with oauth1
Refactored tests to prevent them from using yakbak when they don't need to
Updated swagger version
Update package version by a major number due to the severity of the changes in this PR
Better demos (including using the justified view! `#crossmarketing`)
Updated the README to reflect these changes and best practices

### CODE REVIEW

**Comment by jimwhimpey** [comment]
> Coverage decreased (-5.3%) to 94.328%

WTF @alex-seville !?

File: lib/flickr-transport.js [hunk 1]
```diff
@@ -344,6 +363,19 @@ FlickrTransport.prototype.call = function (definition, params, additionalParams)
 					}
 				});
 
+				responseBody = responseBody || {};
+
+				// When we're requesting a token, it returns text that we need to parse
+				if (res.type === 'text/plain' && (operationId === "getRequestToken" || operationId === "getAccessToken")) {
```
Comment (jeremyruppel) [inline lib/flickr-transport.js]: Are there any other `text/plain` operations? It might be nice to teach superagent how to handle `text/plain` with a [custom response body parser](http://visionmedia.github.io/superagent/#parsing-response-bodies).

File: lib/sdk/auth.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		authenticateUser: function (requestToken, requestTokenSecret, requestTokenVerifier) {
+
+			return new Promise(function (resolve, reject) {
+
+				var requestTokenDefinition = request.sdk.findDefinition(apiV1, 'getRequestToken');
+				var requestURL = requestTokenDefinition.schema.schemes[0] + '://' + requestTokenDefinition.schema.host + requestTokenDefinition.schema.basePath + '/oauth/authorize?oauth_token=' + requestToken;
+
+				if (!process.browser) {
+					if (!requestTokenVerifier) {
+						return new Error('You must specify the request token verifier.');
```
Comment (jeremyruppel) [inline lib/sdk/auth.js]: `reject` instead of `return`?

File: package.json [hunk 3]
```diff
@@ -11,7 +11,8 @@
   ],
   "main": "flickr-sdk.js",
   "browser": {
-    "flickr-js-sdk": "./build/flickr-js-sdk.min.js"
+    "flickr-js-sdk": "./build/flickr-js-sdk.min.js",
+    "justified-layout": "./node_modules/justified-layout/lib/index.js"
```
Comment (jeremyruppel) [inline package.json]: Why is this necessary?

File: lib/flickr-transport.js (hunk 1 above)
Comment (alex-seville) [inline lib/flickr-transport.js]: Nope, just these, they are special

File: lib/sdk/auth.js (hunk 2 above)
Comment (alex-seville) [inline lib/sdk/auth.js]: Good call

File: package.json (hunk 3 above)
Comment (alex-seville) [inline package.json]: It isn't

**Comment by jeremyruppel** [comment]
Seems cromulent! 👍

**Comment by superic** [comment]
👍 👍 👍 👍 👍
